theseus/core/tests/test_objective.py

```diff
@@ -280,6 +280,38 @@ def _check_error_for_data(v1_data_, v2_data_, error_, error_norm_2_):
         assert error.shape == (batch_size, 2 * dof)
         _check_error_for_data(v1_data, v2_data, error, error_norm_2)
 
+        # For input_data and also_update
+
+        v1_data = torch.ones(batch_size, dof) * f1 * 0.1
+        v2_data = torch.ones(batch_size, dof) * f2 * 0.1
+
+        input_data = {"v1": v1_data, "v2": v2_data}
+        error = objective.error(input_data=input_data, also_update=False)
+        error_norm_2 = objective.error_squared_norm(
+            input_data=input_data, also_update=False
+        )
+
+        # To check if variables are not updated
+        assert not torch.equal(objective.optim_vars["v1"].data, input_data["v1"])
```

Comment:
We want to verify if the objects are same or not for the compute graph, right @luisenp?
-> `objective.optim_vars["v1"] is not input_data["v1"]`

Reply:
Also just realized these assert are happening after `objective.error_squared_norm` so technically we are not checking `objective.error` yet.

Reply:
I think the safest is to use new tensor variables in line 285-286, and then check here that `objective.optim_vars["v1"] is v1_data` and that `v1_data is not v1_data_new`.